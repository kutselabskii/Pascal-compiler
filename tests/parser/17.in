program abc;

function abc():integer;
begin

end

var a: integer;

begin
	a := abc();
end.
