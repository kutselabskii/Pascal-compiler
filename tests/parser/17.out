|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- var
        |   |   |-- a
        |   |   |   |__  : integer
        |   |-- abc	//Function_declaration
        |   |   |--  : integer
        |   |   |-- //Arguments list
        |   |   |__ //Block
        |   |       |-- //Declaration part
        |   |       |__ //Statement part
        |   |           |__ //Statement sequence
        |__ //Statement part
            |__ //Statement sequence
                |-- //Assignment_statement
                |   |-- a
                |   |   |__  : integer
                |   |__ abc//Function call
                |       |--  : integer
